authors/apps/articles/tests/test_articles.py

```diff
@@ -444,3 +449,49 @@ def test_read_article(self):
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertTrue(ReadingStat.objects.all().filter(
             user=user.username, articles=article.slug))
+
+    def test_report_article_successfully(self):
+        """Test if the user can an report an article"""
+        self.client.credentials(
+            HTTP_AUTHORIZATION='Bearer ' + self.get_user_token())
+        response = self.client.post(
+            self.url, data=self.article, format='json')
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        slug = response.data.get('slug')
+        report_article_url = reverse("article:report_article", args=[slug])
+        get_response = self.client.post(
+            report_article_url, data=self.report_reason, format='json')
+        self.assertEqual(get_response.status_code,  status.HTTP_201_CREATED)
+
+    def test_report_same_article_twice(self):
+        """Test if the user can an report an article twice"""
+        self.client.credentials(
+            HTTP_AUTHORIZATION='Bearer ' + self.get_user_token())
+        response = self.client.post(
+            self.url, data=self.article, format='json')
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        slug = response.data.get('slug')
+        report_article_url = reverse("article:report_article", args=[slug])
+        self.client.post(report_article_url,
+                         data=self.report_reason, format='json')
+
+        report_article_url2 = reverse("article:report_article", args=[slug])
+        get_response = self.client.post(
+            report_article_url2, data=self.report_reason, format='json')
+        self.assertEqual(get_response.status_code,  status.HTTP_200_OK)
+        self.assertEqual(get_response.data,
+                         'You have already reported the article')
+
+    def test_get_reported_article_when_your_not_admin(self):
+        """Test if the non admin user can report an article"""
```

Comment:
@VivianDoreen The doc string of this test seems to be wrong. What is actually being tested is whether a non admin user can get reported articles.

Reply:
@Opio-Emmanuel-Omona ,Am done with the changes. Kindly review to confirm